src/benchmarks/micro/Serializers/Xml_FromStream.cs

```diff
@@ -63,9 +65,35 @@ public T DataContractSerializer_()
             return (T)dataContractSerializer.ReadObject(memoryStream);
         }
 
-        // YAXSerializer is not included in the benchmarks because it does not allow to deserialize from stream (only from file and string)
+        [GlobalSetup(Target = nameof(DataContractSerializer_BinaryXml_))]
+        public void SetupDataContractSerializer_BinaryXml_()
+        {
+            value = DataGenerator.Generate<T>();
+            memoryStream = new MemoryStream(capacity: short.MaxValue);
+            memoryStream.Position = 0;
+            dataContractSerializer = new DataContractSerializer(typeof(T));
+            using (XmlDictionaryWriter writer = XmlDictionaryWriter.CreateBinaryWriter(memoryStream, null, null, ownsStream: false))
+               dataContractSerializer.WriteObject(writer, value);
+
+            memoryBytes = memoryStream.ToArray();
+            xmlDictionaryReader = XmlDictionaryReader.CreateBinaryReader(memoryBytes, XmlDictionaryReaderQuotas.Max);
+        }
+
+        [BenchmarkCategory(Categories.Libraries)]
+        [Benchmark(Description = nameof(XmlDictionaryReader))]
+        public T DataContractSerializer_BinaryXml_()
+        {
+            ((IXmlBinaryReaderInitializer)xmlDictionaryReader).SetInput(memoryBytes, 0, memoryBytes.Length, null, XmlDictionaryReaderQuotas.Max, null, null);
```

Comment:
>xmlDictionaryReader).SetInput [](http://example.com/codeflow?start=42&length=29)

Consider using [IterationSetup] for this step.

Reply:
I did so earlier for my own benchmarks but that was not very successful for this case.

I think it might be fair game to benchmark with the SetInput call since that is needed for any real world scenario (corewcf , OpenRiaServices etc) where you want to read one "xml document" at a time but pool readers/writers for reuse.

When using iterationsetup the method itself should be quite long running (i think either above guidance or some other guidance says 200ms+)  so it must process lots of objects. 

However I have prepared an alternative version at https://github.com/Daniel-Svensson/dotnet_performance/tree/remote_setoutput with commit 
https://github.com/dotnet/performance/commit/128fcdfac56bdf3b42f26b754dc8ce64de05d2c4 if you rather want that one. It will instead read N (100 but I am open for any value) similar objects inside a single "root" element. So the overhead will go from <= 1% to maybe 0,01% of the execution time.

Reply:
@danmoseley Do you prefer the current version which mimics the end user scenario or should the binary xml version of the benchmarks serialize many more objects than the other benchmark versions ?

unless the ci-failures are related to the changes here I think this is ready for possible merge.

Reply:
This is @StephenMolloy  area to answer..